lerobot/scripts/control_sim_robot.py

```diff
@@ -183,8 +183,14 @@ def record(
     resume: bool = False,
     local_files_only: bool = False,
     run_compute_stats: bool = True,
+    assign_rewards: bool = False,
```

Comment:
Provide `assign` rewards for the simulator.

Reply:
The manually assigned reward doesn't appear to be getting set. I think you'll need something like to overwrite the reward from the gym environment

```
            observation, reward, terminated, _, info = env.step(action)

            success = info.get("is_success", False)
            env_timestamp = info.get("timestamp", dataset.episode_buffer["size"] / fps)

            frame = {
                "action": torch.from_numpy(action),
                "next.reward": reward,
                "next.success": success,
                "seed": seed,
                "timestamp": env_timestamp,
            }

            # Overwrite environment reward with manually assigned reward
            if assign_rewards:
                frame["next.reward"] = events["next.reward"]

                # Should success always be false to match what we do in control_utils?
                frame["next.success"] = False
```

Reply:
I recorded a 2 episode dataset with the above change and next.reward looks correct https://huggingface.co/datasets/jackvial/sim_assign_rewards_test_27

Reply:
To get recording in the simulator working end to end I had to make the following changes
1. 
```
       // control_sim_robot.py
        dataset = LeRobotDataset.create(
            repo_id,
            fps,
            root=root,
            features=features,
            use_videos=video,
            image_writer_processes=num_image_writer_processes,
            image_writer_threads=num_image_writer_threads_per_camera * num_cameras,
            tolerance_s=0.05 # Increased the threshold to prevent check_timestamps_sync from throwing
        )
```

2. Pulled in the create_lerobot_dataset_card fixes from https://github.com/huggingface/lerobot/pull/590/files

Reply:
Ir's great, I have fixed the issue.